aziotctl/aziotctl-common/src/set_log_level.rs

```diff
@@ -0,0 +1,57 @@
+use std::fs;
+use std::io::prelude::*;
+use std::process::Command;
+
+use anyhow::{Context, Result};
+
+use crate::{program_name, ServiceDefinition};
+
+#[allow(clippy::missing_errors_doc)]
+pub fn set_log_level(services: &[&ServiceDefinition], level: log::Level) -> Result<()> {
+    for service in services.iter().map(|s| s.service) {
+        write_log_level_file(service, level)?;
+    }
+    Command::new("systemctl").arg("daemon-reload").output()?;
+
+    println!("Set log level to {} for all services. Run the `{} system restart` command for the changes to take effect.", level, program_name());
+    Ok(())
+}
+
+fn write_log_level_file(service: &str, level: log::Level) -> Result<()> {
+    let directory = format!("/etc/systemd/system/{}.d", service);
+    fs::create_dir_all(&directory)?;
+
+    let filename = format!("{}/log-level.conf", directory);
+    let contents = format!(
+        "[Service]
+Environment=AZIOT_LOG={}",
+        level
+    );
+
+    let mut file = fs::File::create(filename)
+        .with_context(|| format!("Failed to create log-level.conf file for {}", service))?;
```

Comment:
This one context doesn't really help when all the other `Result`s are `?`'d without a context. In any case, if you make the change in the caller of this function as I suggested, using a context within this function becomes unnecessary.